### ISSUE TITLE
SELECT LISTAGG(x) WITHIN GROUP (...) triggers parsing errors in Trino dialect

### ISSUE THREAD

**Issue by rileymcdowell** [sqlfluff/sqlfluff#5583]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I wrote some Trino SQL with a `LISTAGG(x, ',') WITHIN GROUP (ORDER BY x)` aggregate function. This generated a parsing error.

Upon further investigation, I found that `LISTAGG(x, ',')` alone was parsable. This problem seems to be related to the `WITHIN GROUP (ORDER BY x)` modifier that controls ordering in the aggreation. Unfortunately, `LISTAGG(x, ',')` alone is not valid TRINO sql. The `WITHIN GROUP...` component is mandatory query syntax.

### Expected Behaviour

I expect `LISTAGG(x, ',') WITHIN GROUP (ORDER BY x)` to be parsable, and that syntax should be allowable with the default linting configuration for the trino dialect.

### Observed Behaviour

Here's an example SQL file. Aggregate functions require a table, so there's some noise at the top with a bunch of `--noqa` comments that create an in-memory temp table. That can be ignored.

```sql
WITH t AS (                -- noqa
        SELECT * FROM (    -- noqa
            VALUES         -- noqa
                ('b'),     -- noqa
                ('c'),     -- noqa
                ('a')      -- noqa
        ) AS t (v)         -- noqa
)                          -- noqa

SELECT LISTAGG(v, ',') WITHIN GROUP (ORDER BY v)
FROM t
```
The output of this script should be `a,b,c`, so the values from the temp table in sorted order.

Here's my SQLfluff command:

```bash
sqlfluff lint --dialect trino ./query.sql
```

And here's the output.

```txt
== [query.sql] FAIL
L:  11 | P:  24 | AL02 | Implicit/explicit aliasing of columns.
                       | [aliasing.column]
L:  11 | P:  24 | CP02 | Unquoted identifiers must be consistently lower case.
                       | [capitalisation.identifiers]
L:  11 | P:  24 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
L:  11 | P:  31 |  PRS | Line 11, Position 31: Found unparsable section: 'GROUP
                       | (ORDER BY v)'
WARNING: Parsing errors found and dialect is set to 'trino'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

The following query parses fine, indicating the `WITHIN GROUP (ORDER BY v)` component is the problem.

```sql
WITH t AS (                -- noqa
        SELECT * FROM (    -- noqa
            VALUES         -- noqa
                ('b'),     -- noqa
                ('c'),     -- noqa
                ('a')      -- noqa
        ) AS t (v)         -- noqa
)                          -- noqa

SELECT LISTAGG(v, ',') --WITHIN GROUP (ORDER BY v)
FROM t
```

However, `LISTAGG(v, ',')` is not valid TRINO SQL. The `WITHIN GROUP...` component is a mandatory component of the `LISTAGG` syntax.

### How to reproduce

Create a one-line SQL file called query.sql with the following contents:

```sql
WITH t AS (                -- noqa
        SELECT * FROM (    -- noqa
            VALUES         -- noqa
                ('b'),     -- noqa
                ('c'),     -- noqa
                ('a')      -- noqa
        ) AS t (v)         -- noqa
)                          -- noqa

SELECT LISTAGG(v, ',') WITHIN GROUP (ORDER BY v)
FROM t
```

Then run the following command with the latest SQLFluff release:

```bash
sqlfluff lint --dialect trino ./query.sql
```

### Dialect

Trino

### Version

```bash
sqlfluff --version
sqlfluff, version 2.3.5
```

```bash
python --version
Python 3.9.16
```

### Configuration

No configuration file was used. This can be triggered with the default configuration.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by rileymcdowell** [comment]
This issue was fixed by https://github.com/sqlfluff/sqlfluff/issues/5367 in release 3.0.0.a1
